packages/flutter_bloc/lib/src/bloc_provider.dart

```diff
@@ -34,7 +34,7 @@ mixin BlocProviderSingleChildWidget on SingleChildWidget {}
 /// ```
 ///
 /// {@endtemplate}
-class BlocProvider<T extends BlocBase<Object?>>
+class BlocProvider<T extends StateStreamable<Object?>>
```

Comment:
I am not very sure about this suggestion, but maybe be it will be appropriate to change constraints from `T extends StateStreamable<Object?>` to `T extends StateStreamable<Object?>, Closable`

Reply:
If you're suggesting this syntax will imply that T extends StateStreamable and Closable then I don't think this will work.  As far as I know, Dart does not have intersection types.

Reply:
You can extend from a single class only indeed.

Reply:
> If you're suggesting this syntax will imply that T extends StateStreamable and Closable then I don't think this will work. As far as I know, Dart does not have intersection types.

Oh, I'm so sorry for my wording and the misunderstanding caused! 

What I meant, is to create another interface that combines both `StateStreamable<Object?>` and `Closable` in one, something like

```dart
abstract class StateStreamableSource<T extends Object?> implements StateStreamable<T>, Closable { }
```